packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncSession.kt

```diff
@@ -137,6 +142,15 @@ public interface SyncSession {
         progressMode: ProgressMode,
     ): Flow<Progress>
 
+    /**
+     * Create a [Flow] of [ConnectionStateChange]-events to receive notifications of updates to the
+     * session's connection state.
+     *
+     * The flow has an internal buffer of [Channel.BUFFERED] but if the consumer fails to consume
+     * the elements in a timely manner the flow will be completed with an [IllegalStateException].
+     */
+    public fun connectionState(): Flow<ConnectionStateChange>
```

Comment:
If feels a bit akward to have both a `connectionState` property and a `connectionState()` method, but I'm not sure I have a better idea.

Reply:
Yeah #naming 😞 ... Maybe we should follow the pattern that I suggested in https://github.com/realm/realm-kotlin/pull/1118#discussion_r1017519584. Progress flows are not released yet, so we could switch to `progressAsFlow()` and `connectionStateAsFlow` if we want?

Reply:
Updated it to `xAsFlow` after internal discussions.